functions/webhooks/twitter/TwitterToFlex.ts

```diff
@@ -74,12 +75,43 @@ const isValidTwitterPayload = (event: Body, consumerSecret: string): boolean =>
 /**
  * Retrieves a channel by sid or uniqueName
  */
-const retrieveChannel = (context: Context<EnvVars>, uniqueName: string) =>
-  context
+const retrieveChannel = async (
+  context: Context<EnvVars>,
+  uniqueSenderName: string,
+): Promise<string | undefined> => {
+  try {
+    const userChannelMap = await context
+      .getTwilioClient()
+      .sync.services(context.SYNC_SERVICE_SID)
+      .documents(uniqueSenderName)
+      .fetch();
+    console.log('userChannelMap: ', userChannelMap.data);
+
+    return userChannelMap.data.activeChannelSid;
+  } catch (err) {
+    return undefined;
+  }
+};
+
+/**
+ * Updates the user channel map to contain the new channel assigned for this user
+ */
+const createUserChannelMap = async (
+  context: Context<EnvVars>,
+  uniqueSenderName: string,
+  channelSid: string,
+) => {
+  const userChannelMap = await context
     .getTwilioClient()
-    .chat.services(context.CHAT_SERVICE_SID)
-    .channels(uniqueName)
-    .fetch();
+    .sync.services(context.SYNC_SERVICE_SID)
+    .documents.create({
+      data: { activeChannelSid: channelSid },
+      uniqueName: uniqueSenderName,
+      ttl: 259200, // auto romeoved after 3 days
```

Comment:
Typo in `remeoved`. Another question: why 3 days? Is it related to how long we want to keep the task/channel open?

Reply:
Thanks! The 3 days timeout is an arbitrary based on the max time we've been asked a task should live so far. This was one of the questions I had. I believe we need some timeout (in case we miss the "clean up" event), but don't know what would be the correct number. Maybe @nickhurlburt have an opinion?

Reply:
I believe that currently, a task will be timed out after 24 hours.  But it can be set to survive for up to 2 weeks.  If there's no harm to doing so, I'd say make it 2 weeks.  When we add email it'll have a longer TTL, and we won't have to remember to update this.